tracer/src/Datadog.Trace/AppSec/Waf/NativeBindings/WafNative.cs

```diff
@@ -217,30 +187,28 @@ internal IntPtr ObjectMap()
             return ptr;
         }
 
-        internal bool ObjectArrayAdd(IntPtr array, IntPtr entry)
-        {
-            return _objectArrayAddField(array, entry);
-        }
+        internal bool ObjectArrayAdd(IntPtr array, IntPtr entry) => _objectArrayAddField(array, entry);
 
         // Setting entryNameLength to 0 will result in the entryName length being re-computed with strlen
-        internal bool ObjectMapAdd(IntPtr map, string entryName, ulong entryNameLength, IntPtr entry)
-        {
-            return
-                Environment.Is64BitProcess ?
+        internal bool ObjectMapAdd(IntPtr map, string entryName, ulong entryNameLength, IntPtr entry) => Environment.Is64BitProcess ?
                     _objectMapAddFieldX64(map, entryName, entryNameLength, entry) :
                     _objectMapAddFieldX86(map, entryName, (uint)entryNameLength, entry);
-        }
 
-        internal void ObjectFree(IntPtr input)
-        {
-            _objectFreeField(input);
-        }
+        internal void ObjectFreePtr(IntPtr input) => _freeObjectield(input);
+
+        internal void ResultFree(ref DdwafResultStruct output) => _freeResultField(ref output);
+
+        /// <summary>
+        /// internally, waf is also freeing the errors property, so no need to free Errors before
+        /// </summary>
+        /// <param name="output">the rulsetinfo structure</param>
+        internal void RuleSetInfoFree(ref DdwafRuleSetInfoStruct output) => _rulesetInfoFreeField(ref output);
```

Comment:
I know you say that we don't need to use this function, but not sure I understand the logic of you comment, can the WAF free the memory if it's been passed to the C# code to use?

Also, if we don't need to use it, we probably should expose it, unless we're sure we'll use it in the future.

Reply:
It is called by the [`WafConfigurator` ](https://github.com/DataDog/dd-trace-dotnet/blob/297eca7993445321247d43d875f62b42f3c1c724/tracer/src/Datadog.Trace/AppSec/Waf/Initialization/WafConfigurator.cs#L70), my comment was misleading I changed it to:
` /// Only this function needs to be called on DdwafRuleSetInfoStruct, no need to dispose the Errors object inside because waf takes care of it`
because `DdwafRuleSetInfoStruct ` contains a dd waf object and I was wondering if we should take care of that too, but no